### PR TITLE
[MC-1243] Manage Engine Service Desk 1.0.0

### PR DESCRIPTION
**Pull request by mgrandys-r7**
### Description

Describe the proposed changes:

  - Create Manage Engine Service Desk plugin with 14 actions

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Run

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Manage Engine Service Desk:1.0.0. Step name: add_request\nAdding a request...\n",
    "status": "ok",
    "meta": {},
    "output": {
      "request_id": 56,
      "status": "success",
      "status_code": 2000
    }
  },
  "version": "v1",
  "type": "action_event"
}
```

<summary>
docker run --rm -i rapid7/manage_engine_service_desk:1.0.0 run < tests/add_request.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Manage Engine Service Desk:1.0.0. Step name: add_request_note\nAdding note to the request with 17 id...\n",
    "status": "ok",
    "meta": {},
    "output": {
      "request_id": 17,
      "request_note_id": 336,
      "status": "success",
      "status_code": 2000
    }
  },
  "version": "v1",
  "type": "action_event"
}
```

<summary>
docker run --rm -i rapid7/manage_engine_service_desk:1.0.0 run < tests/add_request_note.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Manage Engine Service Desk:1.0.0. Step name: add_resolution\nAdding resolution to the request with 40 id...\n",
    "status": "ok",
    "meta": {},
    "output": {
      "request_id": 40,
      "status": "success",
      "status_code": 2000
    }
  },
  "version": "v1",
  "type": "action_event"
}
```

<summary>
docker run --rm -i rapid7/manage_engine_service_desk:1.0.0 run < tests/add_resolution.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Manage Engine Service Desk:1.0.0. Step name: assign_request\nAssigning a request with 43 id...\n",
    "status": "ok",
    "meta": {},
    "output": {
      "request_id": 43,
      "status": "success",
      "status_code": 2000
    }
  },
  "version": "v1",
  "type": "action_event"
}
```

<summary>
docker run --rm -i rapid7/manage_engine_service_desk:1.0.0 run < tests/assign_request.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Manage Engine Service Desk:1.0.0. Step name: close_request\nClosing a request with 45 id...\n",
    "status": "ok",
    "meta": {},
    "output": {
      "request_id": 45,
      "status": "success",
      "status_code": 2000
    }
  },
  "version": "v1",
  "type": "action_event"
}
```

<summary>
docker run --rm -i rapid7/manage_engine_service_desk:1.0.0 run < tests/close_request.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Manage Engine Service Desk:1.0.0. Step name: delete_request\nDeleting a request with 55 id...\n",
    "status": "ok",
    "meta": {},
    "output": {
      "request_id": 55,
      "status": "success",
      "status_code": 2000
    }
  },
  "version": "v1",
  "type": "action_event"
}
```

<summary>
docker run --rm -i rapid7/manage_engine_service_desk:1.0.0 run < tests/delete_request.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Manage Engine Service Desk:1.0.0. Step name: delete_request_note\nDeleting a note with 327 note id added to the request with 17 id\u2026\n",
    "status": "ok",
    "meta": {},
    "output": {
      "request_id": 17,
      "status": "success",
      "status_code": 2000
    }
  },
  "version": "v1",
  "type": "action_event"
}
```

<summary>
docker run --rm -i rapid7/manage_engine_service_desk:1.0.0 run < tests/delete_request_note.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Manage Engine Service Desk:1.0.0. Step name: edit_request\nEditing a request with 27 id...\n",
    "status": "ok",
    "meta": {},
    "output": {
      "request_id": 27,
      "status": "success",
      "status_code": 2000
    }
  },
  "version": "v1",
  "type": "action_event"
}
```

<summary>
docker run --rm -i rapid7/manage_engine_service_desk:1.0.0 run < tests/edit_request.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Manage Engine Service Desk:1.0.0. Step name: edit_request_note\nUpdating a note with 329 note id added to the request with 17 id...\n",
    "status": "ok",
    "meta": {},
    "output": {
      "request_id": 17,
      "status": "success",
      "status_code": 2000
    }
  },
  "version": "v1",
  "type": "action_event"
}
```

<summary>
docker run --rm -i rapid7/manage_engine_service_desk:1.0.0 run < tests/edit_request_note.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nmorecode/Manage Engine Service Desk:1.0.0. Step name: get_list_request\nGetting list of requests...\n",
    "status": "ok",
    "meta": {},
    "output": {
      "requests": [
        {
          "created_by": {
            "id": 4,
            "name": "administrator",
            "is_vipuser": false
          },
          "created_time": "Aug 9, 2022 06:06 AM",
          "due_by_time": "Aug 9, 2022 05:00 PM",
          "id": 71,
          "is_overdue": true,
          "is_service_request": false,
          "priority": {
            "name": "Low",
            "id": 1
          },
          "requester": {
            "id": 4,
            "name": "administrator",
            "is_vipuser": false
          },
          "site": {
            "name": "Test Site",
            "id": 301
          },
          "status": {
            "name": "Open",
            "id": 2
          },
          "subject": "Plugin action test add request 2",
          "technician": {
            "id": 4,
            "name": "administrator",
            "is_vipuser": false
          }
        },
        {
          "created_by": {
            "id": 4,
            "name": "administrator",
            "is_vipuser": false
          },
          "created_time": "Aug 8, 2022 04:54 AM",
          "due_by_time": "Aug 8, 2022 05:00 PM",
          "id": 61,
          "is_overdue": true,
          "is_service_request": false,
          "priority": {
            "name": "Low",
            "id": 1
          },
          "requester": {
            "id": 4,
            "name": "administrator",
            "is_vipuser": false
          },
          "site": {
            "name": "Test Site",
            "id": 301
          },
          "status": {
            "name": "Open",
            "id": 2
          },
          "subject": "Plugin action test add request 2",
          "technician": {
            "id": 4,
            "name": "administrator",
            "is_vipuser": false
          }
        },
        {
          "created_by": {
            "id": 4,
            "name": "administrator",
            "is_vipuser": false
          },
          "created_time": "Aug 8, 2022 04:41 AM",
          "due_by_time": "Aug 8, 2022 05:00 PM",
          "id": 60,
          "is_overdue": true,
          "is_service_request": false,
          "priority": {
            "name": "Low",
            "id": 1
          },
          "requester": {
            "id": 4,
            "name": "administrator",
            "is_vipuser": false
          },
          "site": {
            "name": "Test Site",
            "id": 301
          },
          "status": {
            "name": "Open",
            "id": 2
          },
          "subject": "Plugin action test add request 2",
          "technician": {
            "id": 4,
            "name": "administrator",
            "is_vipuser": false
          }
        },
        {
          "created_by": {
            "id": 4,
            "name": "administrator",
            "is_vipuser": false
          },
          "created_time": "Jul 13, 2022 01:20 AM",
          "due_by_time": "Jul 15, 2022 05:00 PM",
          "id": 27,
          "is_overdue": true,
          "is_service_request": false,
          "priority": {
            "name": "Low",
            "id": 1
          },
          "requester": {
            "id": 4,
            "name": "administrator",
            "is_vipuser": false
          },
          "site": {
            "name": "Test Site",
            "id": 301
          },
          "status": {
            "name": "Open",
            "id": 2
          },
          "subject": "Plugin action test add request 2",
          "technician": {
            "id": 4,
            "name": "administrator",
            "is_vipuser": false
          }
        }
      ],
      "status": "success"
    }
  },
  "version": "v1",
  "type": "action_event"
}
```

<summary>
docker run --rm -i rapid7/manage_engine_service_desk:1.0.0 run < tests/get_list_request.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Manage Engine Service Desk:1.0.0. Step name: get_list_request_notes\nGetting a list of notes added to the request with 42 id\u2026\n",
    "status": "ok",
    "meta": {},
    "output": {
      "request_id": 42,
      "notes": [
        {
          "last_updated_by": {
            "name": "administrator",
            "id": 4
          },
          "added_time": "Aug 3, 2022 12:55 AM",
          "last_updated_time": "Aug 3, 2022 12:55 AM",
          "added_by": {
            "name": "administrator",
            "id": 4
          },
          "show_to_requester": true
        },
        {
          "last_updated_by": {
            "name": "administrator",
            "id": 4
          },
          "added_time": "Aug 3, 2022 12:55 AM",
          "last_updated_time": "Aug 3, 2022 12:55 AM",
          "added_by": {
            "name": "administrator",
            "id": 4
          },
          "show_to_requester": true
        }
      ],
      "status": "success",
      "status_code": 2000
    }
  },
  "version": "v1",
  "type": "action_event"
}
```

<summary>
docker run --rm -i rapid7/manage_engine_service_desk:1.0.0 run < tests/get_list_request_notes.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Manage Engine Service Desk:1.0.0. Step name: get_request\nGetting a request with 50 id...\n",
    "status": "ok",
    "meta": {},
    "output": {
      "request": {
        "subject": "Plugin action test add request 2",
        "requester": {
          "id": 4,
          "name": "administrator",
          "is_vipuser": false
        },
        "description": "dessssc",
        "request_type": {
          "name": "Incident",
          "id": 1
        },
        "impact": {
          "name": "Medium",
          "id": 2
        },
        "status": {
          "name": "Open",
          "id": 2
        },
        "mode": {
          "name": "Web Form",
          "id": 2
        },
        "level": {
          "name": "Tier 4",
          "id": 4
        },
        "id": 54,
        "urgency": {
          "name": "Urgent",
          "id": 1
        },
        "priority": {
          "name": "Normal",
          "id": 2
        },
        "service_category": {
          "name": "User Management",
          "id": 8
        },
        "assets": [
          {
            "name": "keyboardMC",
            "id": 301
          }
        ],
        "site": {
          "name": "Test Site",
          "id": 301
        },
        "technician": {
          "id": 4,
          "name": "administrator",
          "is_vipuser": false
        },
        "category": {
          "name": "Software",
          "id": 3
        },
        "subcategory": {
          "name": "MS Office",
          "id": 5
        },
        "email_ids_to_notify": [
          "[test@rapid7.com](mailto:test@rapid7.com)"
        ],
        "udf_fields": {},
        "created_time": "Aug 3, 2022 01:11 AM",
        "due_by_time": "Aug 3, 2022 01:00 PM",
        "created_by": {
          "id": 4,
          "name": "administrator",
          "is_vipuser": false
        },
        "is_service_request": false,
        "has_notes": false,
        "is_overdue": true
      },
      "status": "success"
    }
  },
  "version": "v1",
  "type": "action_event"
}
```

<summary>
docker run --rm -i rapid7/manage_engine_service_desk:1.0.0 run < tests/get_request.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Manage Engine Service Desk:1.0.0. Step name: get_resolution\nGetting resolution added to the request with 40 id...\n",
    "status": "ok",
    "meta": {},
    "output": {
      "request_id": 40,
      "content": "asd",
      "status": "success",
      "status_code": 2000
    }
  },
  "version": "v1",
  "type": "action_event"
}
```

<summary>
docker run --rm -i rapid7/manage_engine_service_desk:1.0.0 run < tests/get_resolution.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Manage Engine Service Desk:1.0.0. Step name: pickup_request\nPicking up a request with 40 id...\n",
    "status": "ok",
    "meta": {},
    "output": {
      "request_id": 40,
      "status": "success",
      "status_code": 2000
    }
  },
  "version": "v1",
  "type": "action_event"
}
```

<summary>
docker run --rm -i rapid7/manage_engine_service_desk:1.0.0 run < tests/pickup_request.json
</summary>
</details>

### Validate

<details>

```
[*] Validating plugin at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator CloudReadyConnectionCredentialTokenValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Executing validator VersionPinValidator
[*] Executing validator EncodingValidator
[*] Executing validator ExampleInputValidator
[*] Executing validator CloudReadyValidator
[*] Executing validator SupportedVersionValidator
[*] Executing validator UnapprovedKeywordsValidator
[*] Executing validator HelpExampleValidator
[*] Executing validator Version Increment Validator
[*] Plugin failed validation! The following validation errors occurred:

Validator "RegenerationValidator" failed!

Cause: Error: Hashes between provided plugin and checksum were not equal. Regenerate the plugin and push to working branch.

Validator "HelpExampleValidator" failed!
Cause: Invalid JSON example identified in Example input of action titled Add Request. Please correct the JSON example in help.md
Invalid JSON example identified in Example input of action titled Input. Please correct the JSON example in help.md
Improperly formatted JSON example identified in Example input of action titled Input. Please updated the JSON example in help.md with 2 space indentation.
Invalid JSON example identified in Example input of action titled Input. Please correct the JSON example in help.md
Invalid JSON example identified in Example input of action titled Input. Please correct the JSON example in help.md
Invalid JSON example identified in Example input of action titled Input. Please correct the JSON example in help.md

----

[*] Total time elapsed: 10698.105ms

make: *** [validate] Error 1
```

<summary>
icon-validate --all .
</summary>
</details>

### CODE REVIEW

**Review by dsliwinski-r7 — APPROVED** [review]

**Review by dmandyna-r7 — APPROVED** [review]

File: plugins/manage_engine_service_desk/icon_manage_engine_service_desk/actions/add_request_note/action.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+            description=Component.DESCRIPTION,
+            input=AddRequestNoteInput(),
+            output=AddRequestNoteOutput(),
+        )
+
+    def run(self, params: dict = None) -> dict:
+        request_id = params.get(Input.REQUEST_ID)
+        response_json = self.connection.api_client.add_request_note(
+            request_id=request_id,
+            description=params.get(Input.DESCRIPTION),
```
Comment (jkaluza-rapid7) [inline plugins/manage_engine_service_desk/icon_manage_engine_service_desk/actions/add_request_note/action.py:23]: Could you check if the description is a valid html string? What happens if it's not? Do we get an explicit error from manage engine?

File: plugins/manage_engine_service_desk/plugin.spec.yaml [hunk 2]
```diff
@@ -0,0 +1,1345 @@
...
+name: manage_engine_service_desk
+title: Manage Engine Service Desk
+description: ManageEngine's Service Desk has the ability to centralize and capture reported issues, this allows security/IT admin to track and manage all incidents in an easy manner. The numerous help desk tickets raised are organized and tracked in the Requests module. The Requests module enables you to handle tickets promptly, assign tickets to technicians, merge similar requests, and so on
+version: 1.0.0
+supported_versions: ["ServiceDesk Plus 13008"]
+vendor: rapid7
+support: rapid7
+status: []
+resources:
+  source_url: https://github.com/rapid7/insightconnect-plugins/tree/master/plugins/manage_engine_service_desk
```
Comment (jkaluza-rapid7) [inline plugins/manage_engine_service_desk/plugin.spec.yaml:13]: Could you add API-doc link?

**Review by jkaluza-rapid7 — CHANGES_REQUESTED** [review]
Good job!

File: plugins/manage_engine_service_desk/icon_manage_engine_service_desk/actions/add_request_note/action.py (hunk 1 above)
Comment (mgrandys-r7) [reply plugins/manage_engine_service_desk/icon_manage_engine_service_desk/actions/add_request_note/action.py:23]: Manage Engine is not returning an error if it's not a valid HTML, also user can provide pure string.

File: plugins/manage_engine_service_desk/plugin.spec.yaml (hunk 2 above)
Comment (mgrandys-r7) [reply plugins/manage_engine_service_desk/plugin.spec.yaml:13]: @jkaluza-rapid7 Done, please have a look

File: plugins/manage_engine_service_desk/icon_manage_engine_service_desk/util/helpers.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+                del cleaned_dict[key]
+        elif value in [None, "", 0, [], {}]:
+            del cleaned_dict[key]
+    return cleaned_dict
+
+
+def remove_other_keys(input_dict: dict, keys_to_keep: list) -> dict:
+    if not isinstance(input_dict, dict):
+        return input_dict
+    return {key: input_dict.get(key) for key in keys_to_keep}
```
[the remark this answers is not in the document]
Comment (mgrandys-r7) [reply plugins/manage_engine_service_desk/icon_manage_engine_service_desk/util/helpers.py:51]: @jkaluza-rapid7 Done, please have a look

**Review by jkaluza-rapid7 — APPROVED** [review]

**Review by tharward-r7 — CHANGES_REQUESTED** [review]
The review comments here might be out of date since the branch was updated; submitting to continue review. Please take a look at the language suggestions here.

File: plugins/manage_engine_service_desk/help.md [hunk 4]
```diff
@@ -0,0 +1,1117 @@
...
+{
+  "request_id": 55,
+  "status": "success",
+  "status_code": 2000
+}
+```
+
+#### Get List Request
+
+This action lets you view the details of all the requests.
```
Comment (tharward-r7) [inline plugins/manage_engine_service_desk/help.md:473]:
```suggestion
View the details of a list of requests matching a search.
```

File: plugins/manage_engine_service_desk/help.md [hunk 5]
```diff
@@ -0,0 +1,1117 @@
...
+      }
+    }
+  ],
+  "status": "success"
+}
+```
+
+#### Get List Request Notes
+
+This action allows you to get all the notes.
```
Comment (tharward-r7) [inline plugins/manage_engine_service_desk/help.md:659]:
```suggestion
Get the list of all notes associated with the given request.
```

File: plugins/manage_engine_service_desk/help.md [hunk 6]
```diff
@@ -0,0 +1,1117 @@
...
+    }
+  ],
+  "status": "success",
+  "status_code": 2000
+}
+```
+
+#### Get Request
+
+This action allows you to view the request details by the unique request id.
```
Comment (tharward-r7) [inline plugins/manage_engine_service_desk/help.md:724]:
```suggestion
View the details of a request given the request id.
```

File: plugins/manage_engine_service_desk/help.md [hunk 7]
```diff
@@ -0,0 +1,1117 @@
...
+    "has_notes": false,
+    "is_overdue": true
+  },
+  "status": "success"
+}
+```
+
+#### Get Resolution
+
+This action allows you to get a resolution of the request.
```
Comment (tharward-r7) [inline plugins/manage_engine_service_desk/help.md:837]:
```suggestion
Get the resolution of the given request.
```

**Review by tharward-r7 — CHANGES_REQUESTED** [review]
Additional language changes mostly for action descriptions. I can approve once these are changed.

File: plugins/manage_engine_service_desk/help.md [hunk 8]
```diff
@@ -0,0 +1,1113 @@
...
+}
+```
+
+## Technical Details
+
+### Actions
+
+#### Add Request
+
+This action allows you to add new a request. Subject and requester parameters are required, others are optional. In every parameter containing `ID` and `Name` fields please provide only one of them.
```
Comment (tharward-r7) [inline plugins/manage_engine_service_desk/help.md:51]:
```suggestion
Add a new request. Subject and requester parameters are required, others are optional. In every parameter containing `ID` and `Name` fields please provide only one of them.
```

File: plugins/manage_engine_service_desk/help.md [hunk 9]
```diff
@@ -0,0 +1,1113 @@
...
+{
+  "request_id": 27,
+  "status": "success",
+  "status_code": 2000
+}
+```
+
+#### Assign Request
+
+This action allows you to assign a request to a technician or group. Request ID is required, as well as at least one of Group or Technician. In every parameter containing `ID` and `Name` fields please provide only one or the other.
```
Comment (tharward-r7) [inline plugins/manage_engine_service_desk/help.md:197]:
```suggestion
Assign a request to a technician or group. Request ID is required, as well as at least one of Group or Technician. In every parameter containing `ID` and `Name` fields please provide only one or the other.
```

File: plugins/manage_engine_service_desk/help.md [hunk 10]
```diff
@@ -0,0 +1,1113 @@
...
+{
+  "request_id": 55,
+  "status": "success",
+  "status_code": 2000
+}
+```
+
+#### Edit Request
+
+This action allows you to update the request by using the unique request ID. At least one parameter other than Request ID is required. In every parameter containing `ID` and `Name` fields please provide only one or the other.
```
Comment (tharward-r7) [inline plugins/manage_engine_service_desk/help.md:361]:
```suggestion
Update the given request. At least one parameter other than Request ID is required. In every parameter containing `ID` and `Name` fields please provide only one or the other.
```

File: plugins/manage_engine_service_desk/help.md [hunk 11]
```diff
@@ -0,0 +1,1113 @@
...
+  "request_id": 27,
+  "content": "asd",
+  "status": "success",
+  "status_code": 2000
+}
+```
+
+#### Pickup Request
+
+This action allows you to pick up a request in your name as a technician.
```
Comment (tharward-r7) [inline plugins/manage_engine_service_desk/help.md:871]:
```suggestion
Pick up (assign) a given request in your name as a technician.
```

**Review by tharward-r7 — APPROVED** [review]
I missed a couple descriptions in the form "This action..." -- update if you can, but not blocking.

**Comment by dsliwinski-r7** [comment]
> I missed a couple descriptions in the form "This action..." -- update if you can, but not blocking.

@tharward-r7 Done :)

**Review by tharward-r7 — APPROVED** [review]

**Review by blaxminarayan-r7 — CHANGES_REQUESTED** [review]

**Review by blaxminarayan-r7 — APPROVED** [review]

**Pull request merged**